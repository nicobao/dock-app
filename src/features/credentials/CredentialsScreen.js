import React, {useEffect} from 'react';
import {translate} from 'src/locales';
import {PolkadotIcon} from '../../components/PolkadotIcon';
import {
  Box,
  EmptyCredentialIcon,
  Header,
  IconButton,
  NBox,
  ScreenContainer,
  Theme,
  Typography,
  DotsVerticalIcon,
} from '../../design-system';
import PlusCircleWhiteIcon from '../../assets/icons/plus-circle-white.svg';
import {addTestId} from '../../core/automation-utils';
import {
  Center,
  Image,
  Text,
  Stack,
  Menu,
  Pressable,
  FlatList,
} from 'native-base';
import {useCredentials, getDIDAddress} from './credentials';
import {formatDate} from '@docknetwork/wallet-sdk-core/lib/core/format-utils';
import {withErrorBoundary} from 'src/core/error-handler';
import {View} from 'react-native';
import {CredentialStatus} from './components/CredentialStatus';
import {useIsFocused} from '@react-navigation/native';

function shouldRenderAttr(attr) {
  return attr.property !== 'id' && attr.property !== 'title';
}

export function renderObjectAttributes({attributes}) {
  return (
    <>
      {attributes.map((attr, index) => {
        return (
          shouldRenderAttr(attr) && (
            <Stack mb={1} key={`${index}_${attr.value}`}>
              <Text
                textTransform="capitalize"
                fontSize={'12px'}
                fontWeight={600}
                fontFamily={Theme.fontFamily.montserrat}>
                {attr.name}
              </Text>
              <Text fontSize={'12px'}>{attr.value}</Text>
            </Stack>
          )
        );
      })}
    </>
  );
}

export function EmptyCredentials(props) {
  return (
    <Center {...props}>
      <Box
        borderRadius={72}
        width={72}
        height={72}
        backgroundColor={Theme.colors.iconBackgroundColor}>
        <Center h="100%" width="100%">
          <EmptyCredentialIcon color={Theme.colors.description} />
        </Center>
      </Box>
      <Text
        color={Theme.colors.textHighlighted}
        fontSize={14}
        pt={2}
        fontWeight={400}
        fontFamily={Theme.fontFamily.default}>
        {translate('credentials.empty_items')}
      </Text>
    </Center>
  );
}

export const CredentialListItem = withErrorBoundary(
  ({credential, formattedData, credentialActions = <NBox />}) => {
    const {title = translate('credentials.default_title')} = formattedData;

    return (
      <NBox
        backgroundColor={Theme.colors.cardItemBackground}
        p={4}
        borderRadius={10}
        m={2}
        key={`${credential.id}`}>
        <Stack direction="row">
          <Stack>
            <View
              style={{
                width: '98%',
              }}>
              <Text
                mb={4}
                fontSize={'16px'}
                fontWeight={600}
                fontFamily={Theme.fontFamily.montserrat}>
                {title}
              </Text>
            </View>
            {formattedData.humanizedType && (
              <Text
                mt={1}
                mb={2}
                fontSize={'12px'}
                fontWeight={500}
                fontFamily={Theme.fontFamily.montserrat}>
                {formattedData.humanizedType}
              </Text>
            )}

            {renderObjectAttributes(formattedData)}
          </Stack>
          <NBox flex={1} alignItems="flex-end">
            {credentialActions}
          </NBox>
        </Stack>

        <NBox mt={4} flexDirection="row" alignItems={'flex-end'}>
          <NBox>
            <Typography variant={'credentialIssuanceDate'}>
              {formatDate(formattedData.issuanceDate)}
            </Typography>
          </NBox>
          <CredentialStatus credential={credential} />
          <NBox flex={1} alignItems={'flex-end'}>
            {formattedData.image ? (
              <View
                style={{
                  width: 50,
                  height: 50,
                }}>
                <Image
                  borderRadius={8}
                  style={{
                    width: '100%',
                    height: '100%',
                    resizeMode: 'contain',
                  }}
                  alt={title}
                  source={{
                    uri: formattedData.image,
                  }}
                />
              </View>
            ) : getDIDAddress(credential.issuer) ? (
              <PolkadotIcon
                address={getDIDAddress(credential.issuer)}
                size={32}
              />
            ) : (
              <View
                style={{
                  width: 40,
                  height: 40,
                }}>
                <Image
                  borderRadius={8}
                  style={{
                    width: '100%',
                    height: '100%',
                    resizeMode: 'contain',
                  }}
                  alt={title}
                  source={require('../../assets/circle.png')}
                />
              </View>
            )}
          </NBox>
        </NBox>
      </NBox>
    );
  },
);

export function CredentialsScreen({
  credentials,
  onRemove,
  onAdd,
  refreshing,
  onRefresh,
}) {
  return (
    <ScreenContainer {...addTestId('CredentialsScreen')} showTabNavigation>
      <Header>
        <Box
          marginLeft={22}
          marginRight={22}
          flexDirection="row"
          alignItems="center">
          <Box flex={1}>
            <Typography fontFamily="Montserrat" fontSize={24} fontWeight="600">
              {translate('credentials.title')}
            </Typography>
          </Box>
          <Box row>
            <IconButton onPress={onAdd} col>
              <PlusCircleWhiteIcon
                style={{
                  color: Theme.colors.headerIconColor,
                }}
              />
            </IconButton>
          </Box>
        </Box>
      </Header>
      <FlatList
        refreshing={refreshing}
        onRefresh={onRefresh}
        data={credentials}
        renderItem={({item}) => {
          const credentialActions = (
            <Menu
              bg={Theme.colors.tertiaryBackground}
              trigger={triggerProps => {
                return (
                  <Pressable
                    p={2}
                    {...triggerProps}
                    _pressed={{
                      opacity: Theme.touchOpacity,
                    }}>
                    <DotsVerticalIcon />
                  </Pressable>
                );
              }}>
              <Menu.Item onPress={() => onRemove(item)}>
                {translate('account_list.delete_account')}
              </Menu.Item>
            </Menu>
          );
          return (
            <CredentialListItem
              key={item.id}
              credential={item.content}
              formattedData={item.formattedData}
              credentialActions={credentialActions}
            />
          );
        }}
      />
    </ScreenContainer>
  );
}

export function CredentialsContainer(props) {
  const {credentials, handleRemove, onAdd, refreshing, onRefresh} =
    useCredentials();

  return (
    <CredentialsScreen
      credentials={credentials}
      onRemove={handleRemove}
      onAdd={onAdd}
      refreshing={refreshing}
      onRefresh={onRefresh}
    />
  );
}
