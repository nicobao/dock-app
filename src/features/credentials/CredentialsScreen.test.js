import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CredentialsScreen, CredentialListItem} from './CredentialsScreen';

const mockStore = configureMockStore();

const CredentialMocks = {
  EmployeeIDCredential: {
    data: {
      '@context': [
        'https://www.w3.org/2018/credentials/v1',
        {
          dk: 'https://ld.dock.io/credentials#',
          EmployeeIDCredential: 'dk:EmployeeIDCredential',
          name: 'dk:name',
          email: 'dk:email',
          employeeId: 'dk:employeeId',
        },
      ],
      id: 'https://creds-testnet.dock.io/191ce57b4b9195d823937c016e31544c225279e4fb3e7d5ed88038acf5daac42',
      type: ['VerifiableCredential', 'EmployeeIDCredential'],
      credentialSubject: {
        name: 'Alice Doe',
        email: 'alice@dock.io',
        employeeId: 123456,
      },
      issuanceDate: '2022-08-29T15:24:47.799Z',
      name: 'Acme Employee ID',
      proof: {
        type: 'Ed25519Signature2018',
        created: '2022-08-29T16:24:47Z',
        verificationMethod:
          'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA#keys-1',
        proofPurpose: 'assertionMethod',
        proofValue:
          'zYetMX6EPE1Lzqe92ZZopqFQrwN1g9wwXE2UHW6y69cDZQfoCkz7hMZDxURvo8fVW97p2f6tpAfWKEQU8RvRNPxC',
      },
      issuer: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
    },
    formattedData: {
      humanizedType: 'Employee I D Credential',
      title: 'Acme Employee ID',
      subjectName: 'Alice Doe',
      issuerName: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
      date: 'August 29, 2022',
      image: null,
      images: {
        issuerImage: null,
        subjectImage: null,
        mainImage: null,
        imagesList: [],
      },
      documents: [[]],
      template: 'credential',
      qrImage: false,
      attributes: [
        {name: 'Name', property: 'name', value: 'Alice Doe'},
        {name: 'Email', property: 'email', value: 'alice@dock.io'},
        {name: 'Employee Id', property: 'employeeId', value: 123456},
      ],
      subjects: [
        {name: 'Alice Doe', email: 'alice@dock.io', employeeId: 123456},
      ],
      issuer: 'did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA',
      issuanceDate: '2022-08-29T15:24:47.799Z',
      expirationDateStr: '',
      dateStr: 'August 29, 2022',
    },
  },
  UniversityDegreeCredential: {
    data: {
      '@context': [
        'https://www.w3.org/2018/credentials/v1',
        {
          dk: 'https://ld.dock.io/credentials#',
          UniversityDegreeCredential: 'dk:UniversityDegreeCredential',
          degree: 'dk:degree',
          name: 'dk:name',
          dateEarned: 'dk:dateEarned',
          person: 'dk:person',
          fullName: 'dk:fullName',
          dateOfBirth: 'dk:dateOfBirth',
          description: 'dk:description',
          logo: 'dk:logo',
        },
      ],
      id: 'https://creds.dock.io/8e02c35ae370b02f47d7faaf41cb1386768fc75c9fca7caa6bb389dbe61260eb',
      type: ['VerifiableCredential', 'UniversityDegreeCredential'],
      credentialSubject: {
        degree: {
          name: 'Degree Name',
          type: 'Degree Type',
          dateEarned: '2000-09-24',
        },
        person: {fullName: 'Full Name', dateOfBirth: '2000-09-24'},
        id: 'john',
      },
      issuanceDate: '2022-06-27T11:08:30.675Z',
      expirationDate: '2029-06-26T23:00:00.000Z',
      proof: {
        type: 'Ed25519Signature2018',
        created: '2022-06-27T12:09:34Z',
        verificationMethod:
          'did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N#keys-1',
        proofPurpose: 'assertionMethod',
        proofValue:
          'zVVh5QYsUJR3GNBCKTU3dc2JknTQTrTcXqNJWFCcUb7iGx9iaRZgMUmkJhUF9nRaF5tqz6B2xjnzyWF9p7XNpy4L',
      },
      issuer: {
        name: 'John',
        description: '',
        logo: '',
        id: 'did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N',
      },
    },
    formattedData: {
      humanizedType: 'University Degree Credential',
      title: 'University Degree Credential',
      subjectName: 'Degree Name',
      issuerName: 'John',
      date: 'June 27, 2022',
      image: null,
      images: {
        issuerImage: null,
        subjectImage: null,
        mainImage: null,
        imagesList: [],
      },
      documents: [
        [
          {name: 'Degree Name', type: 'Degree Type', dateEarned: '2000-09-24'},
          {fullName: 'Full Name', dateOfBirth: '2000-09-24'},
        ],
      ],
      template: 'diploma',
      qrImage: false,
      attributes: [
        {name: 'Degree name', property: 'degree name', value: 'Degree Name'},
        {name: 'Degree type', property: 'degree type', value: 'Degree Type'},
        {
          name: 'Degree date Earned',
          property: 'degree dateEarned',
          value: '2000-09-24',
        },
        {
          name: 'Person full Name',
          property: 'person fullName',
          value: 'Full Name',
        },
        {
          name: 'Person date Of Birth',
          property: 'person dateOfBirth',
          value: '2000-09-24',
        },
        {name: 'Id', property: 'id', value: 'osinakayah'},
      ],
      subjects: [
        {
          degree: {
            name: 'Degree Name',
            type: 'Degree Type',
            dateEarned: '2000-09-24',
          },
          person: {fullName: 'Full Name', dateOfBirth: '2000-09-24'},
          id: 'john',
        },
      ],
      issuer: {
        name: 'Ossy',
        description: '',
        logo: '',
        id: 'did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N',
      },
      issuanceDate: '2022-06-27T11:08:30.675Z',
      expirationDate: '2029-06-26T23:00:00.000Z',
      expirationDateStr: 'June 27, 2029',
      dateStr: 'June 27, 2022',
    },
  },
};

describe('CredentialsScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialsScreen credentials={[]} onRemove={jest.fn} onAdd={jest.fn} />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render list of credentials correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialsScreen
        credentials={[CredentialMocks.EmployeeIDCredential.data]}
        onRemove={jest.fn}
        onAdd={jest.fn}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly credential list with string issuer', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialListItem
        credential={CredentialMocks.EmployeeIDCredential.data}
        formattedData={CredentialMocks.EmployeeIDCredential.formattedData}
        onRemove={jest.fn}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly credential list with issuer object', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialListItem
        credential={CredentialMocks.UniversityDegreeCredential.data}
        formattedData={CredentialMocks.UniversityDegreeCredential.formattedData}
        onRemove={jest.fn}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render credential verifier features', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CredentialListItem
        credentialVerifierEnabled={true}
        credential={CredentialMocks.UniversityDegreeCredential.data}
        formattedData={CredentialMocks.UniversityDegreeCredential.formattedData}
        onRemove={jest.fn}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
