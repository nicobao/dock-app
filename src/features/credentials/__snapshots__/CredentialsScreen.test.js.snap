// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialsScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="CredentialsScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="CredentialsScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={22}
          marginRight={22}
        >
          <Box
            flex={1}
          >
            <Styled(Component)
              fontFamily="Montserrat"
              fontSize={24}
              fontWeight="600"
            >
              Credentials
            </Styled(Component)>
          </Box>
          <Box
            row={true}
          >
            <IconButton
              col={true}
              onPress={[Function]}
            >
              <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />
            </IconButton>
          </Box>
        </Box>
      </Styled(View)>
      <EmptyCredentials
        mt="50%"
      />
      <ForwardRef
        data={Array []}
        renderItem={[Function]}
      />
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`CredentialsScreen should render correctly credential list with issuer object 1`] = `
<Component
  credential={
    Object {
      "@context": Array [
        "https://www.w3.org/2018/credentials/v1",
        Object {
          "UniversityDegreeCredential": "dk:UniversityDegreeCredential",
          "dateEarned": "dk:dateEarned",
          "dateOfBirth": "dk:dateOfBirth",
          "degree": "dk:degree",
          "description": "dk:description",
          "dk": "https://ld.dock.io/credentials#",
          "fullName": "dk:fullName",
          "logo": "dk:logo",
          "name": "dk:name",
          "person": "dk:person",
        },
      ],
      "credentialSubject": Object {
        "degree": Object {
          "dateEarned": "2000-09-24",
          "name": "Degree Name",
          "type": "Degree Type",
        },
        "id": "john",
        "person": Object {
          "dateOfBirth": "2000-09-24",
          "fullName": "Full Name",
        },
      },
      "expirationDate": "2029-06-26T23:00:00.000Z",
      "id": "https://creds.dock.io/8e02c35ae370b02f47d7faaf41cb1386768fc75c9fca7caa6bb389dbe61260eb",
      "issuanceDate": "2022-06-27T11:08:30.675Z",
      "issuer": Object {
        "description": "",
        "id": "did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N",
        "logo": "",
        "name": "John",
      },
      "proof": Object {
        "created": "2022-06-27T12:09:34Z",
        "proofPurpose": "assertionMethod",
        "proofValue": "zVVh5QYsUJR3GNBCKTU3dc2JknTQTrTcXqNJWFCcUb7iGx9iaRZgMUmkJhUF9nRaF5tqz6B2xjnzyWF9p7XNpy4L",
        "type": "Ed25519Signature2018",
        "verificationMethod": "did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N#keys-1",
      },
      "type": Array [
        "VerifiableCredential",
        "UniversityDegreeCredential",
      ],
    }
  }
  formattedData={
    Object {
      "attributes": Array [
        Object {
          "name": "Degree name",
          "property": "degree name",
          "value": "Degree Name",
        },
        Object {
          "name": "Degree type",
          "property": "degree type",
          "value": "Degree Type",
        },
        Object {
          "name": "Degree date Earned",
          "property": "degree dateEarned",
          "value": "2000-09-24",
        },
        Object {
          "name": "Person full Name",
          "property": "person fullName",
          "value": "Full Name",
        },
        Object {
          "name": "Person date Of Birth",
          "property": "person dateOfBirth",
          "value": "2000-09-24",
        },
        Object {
          "name": "Id",
          "property": "id",
          "value": "osinakayah",
        },
      ],
      "date": "June 27, 2022",
      "dateStr": "June 27, 2022",
      "documents": Array [
        Array [
          Object {
            "dateEarned": "2000-09-24",
            "name": "Degree Name",
            "type": "Degree Type",
          },
          Object {
            "dateOfBirth": "2000-09-24",
            "fullName": "Full Name",
          },
        ],
      ],
      "expirationDate": "2029-06-26T23:00:00.000Z",
      "expirationDateStr": "June 27, 2029",
      "humanizedType": "University Degree Credential",
      "image": null,
      "images": Object {
        "imagesList": Array [],
        "issuerImage": null,
        "mainImage": null,
        "subjectImage": null,
      },
      "issuanceDate": "2022-06-27T11:08:30.675Z",
      "issuer": Object {
        "description": "",
        "id": "did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N",
        "logo": "",
        "name": "Ossy",
      },
      "issuerName": "John",
      "qrImage": false,
      "subjectName": "Degree Name",
      "subjects": Array [
        Object {
          "degree": Object {
            "dateEarned": "2000-09-24",
            "name": "Degree Name",
            "type": "Degree Type",
          },
          "id": "john",
          "person": Object {
            "dateOfBirth": "2000-09-24",
            "fullName": "Full Name",
          },
        },
      ],
      "template": "diploma",
      "title": "University Degree Credential",
    }
  }
  onRemove={[Function]}
/>
`;

exports[`CredentialsScreen should render correctly credential list with string issuer 1`] = `
<Component
  credential={
    Object {
      "@context": Array [
        "https://www.w3.org/2018/credentials/v1",
        Object {
          "EmployeeIDCredential": "dk:EmployeeIDCredential",
          "dk": "https://ld.dock.io/credentials#",
          "email": "dk:email",
          "employeeId": "dk:employeeId",
          "name": "dk:name",
        },
      ],
      "credentialSubject": Object {
        "email": "alice@dock.io",
        "employeeId": 123456,
        "name": "Alice Doe",
      },
      "id": "https://creds-testnet.dock.io/191ce57b4b9195d823937c016e31544c225279e4fb3e7d5ed88038acf5daac42",
      "issuanceDate": "2022-08-29T15:24:47.799Z",
      "issuer": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "name": "Acme Employee ID",
      "proof": Object {
        "created": "2022-08-29T16:24:47Z",
        "proofPurpose": "assertionMethod",
        "proofValue": "zYetMX6EPE1Lzqe92ZZopqFQrwN1g9wwXE2UHW6y69cDZQfoCkz7hMZDxURvo8fVW97p2f6tpAfWKEQU8RvRNPxC",
        "type": "Ed25519Signature2018",
        "verificationMethod": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA#keys-1",
      },
      "type": Array [
        "VerifiableCredential",
        "EmployeeIDCredential",
      ],
    }
  }
  formattedData={
    Object {
      "attributes": Array [
        Object {
          "name": "Name",
          "property": "name",
          "value": "Alice Doe",
        },
        Object {
          "name": "Email",
          "property": "email",
          "value": "alice@dock.io",
        },
        Object {
          "name": "Employee Id",
          "property": "employeeId",
          "value": 123456,
        },
      ],
      "date": "August 29, 2022",
      "dateStr": "August 29, 2022",
      "documents": Array [
        Array [],
      ],
      "expirationDateStr": "",
      "humanizedType": "Employee I D Credential",
      "image": null,
      "images": Object {
        "imagesList": Array [],
        "issuerImage": null,
        "mainImage": null,
        "subjectImage": null,
      },
      "issuanceDate": "2022-08-29T15:24:47.799Z",
      "issuer": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "issuerName": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
      "qrImage": false,
      "subjectName": "Alice Doe",
      "subjects": Array [
        Object {
          "email": "alice@dock.io",
          "employeeId": 123456,
          "name": "Alice Doe",
        },
      ],
      "template": "credential",
      "title": "Acme Employee ID",
    }
  }
  onRemove={[Function]}
/>
`;

exports[`CredentialsScreen should render list of credentials correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="CredentialsScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="CredentialsScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={22}
          marginRight={22}
        >
          <Box
            flex={1}
          >
            <Styled(Component)
              fontFamily="Montserrat"
              fontSize={24}
              fontWeight="600"
            >
              Credentials
            </Styled(Component)>
          </Box>
          <Box
            row={true}
          >
            <IconButton
              col={true}
              onPress={[Function]}
            >
              <SvgFile
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
              />
            </IconButton>
          </Box>
        </Box>
      </Styled(View)>
      <ForwardRef
        data={
          Array [
            Object {
              "@context": Array [
                "https://www.w3.org/2018/credentials/v1",
                Object {
                  "EmployeeIDCredential": "dk:EmployeeIDCredential",
                  "dk": "https://ld.dock.io/credentials#",
                  "email": "dk:email",
                  "employeeId": "dk:employeeId",
                  "name": "dk:name",
                },
              ],
              "credentialSubject": Object {
                "email": "alice@dock.io",
                "employeeId": 123456,
                "name": "Alice Doe",
              },
              "id": "https://creds-testnet.dock.io/191ce57b4b9195d823937c016e31544c225279e4fb3e7d5ed88038acf5daac42",
              "issuanceDate": "2022-08-29T15:24:47.799Z",
              "issuer": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA",
              "name": "Acme Employee ID",
              "proof": Object {
                "created": "2022-08-29T16:24:47Z",
                "proofPurpose": "assertionMethod",
                "proofValue": "zYetMX6EPE1Lzqe92ZZopqFQrwN1g9wwXE2UHW6y69cDZQfoCkz7hMZDxURvo8fVW97p2f6tpAfWKEQU8RvRNPxC",
                "type": "Ed25519Signature2018",
                "verificationMethod": "did:dock:5D5K67AnGLBKSVJNkFW25uaDa7cSUjx6Zj55tzMdqQvQDPkA#keys-1",
              },
              "type": Array [
                "VerifiableCredential",
                "EmployeeIDCredential",
              ],
            },
          ]
        }
        renderItem={[Function]}
      />
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
